src/core/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,298 @@
+    <nav className={cx('iui-carousel-navigation', className)} {...rest}>
+      <div className='iui-carousel-navigation-left'>
+        <IconButton
+          styleType='borderless'
+          size='small'
+          tabIndex={-1}
+          onClick={() =>
+            setCurrentIndex((old) => (slideCount + old - 1) % slideCount)
+          }
+          data-pressed={keysPressed['ArrowLeft'] || undefined}
+        >
+          <SvgChevronLeft />
+        </IconButton>
+      </div>
+      {children}
+      <div className='iui-carousel-navigation-right'>
+        <IconButton
+          styleType='borderless'
+          size='small'
+          tabIndex={-1}
+          onClick={() =>
+            setCurrentIndex((old) => (slideCount + old + 1) % slideCount)
+          }
+          data-pressed={keysPressed['ArrowRight'] || undefined}
+        >
+          <SvgChevronRight />
+        </IconButton>
+      </div>
+    </nav>
+  );
+};
+
+const CarouselDots = (
+  props: {
+    length?: number;
+    maxCount?: number;
+  } & React.ComponentPropsWithoutRef<'ol'>,
+) => {
+  const { maxCount = 5, className, children, ...rest } = props;
+  const context = React.useContext(CarouselContext);
+  if (!context) {
+    throw new Error('TODO: this needs to be usable without context');
+  }
+
+  const { slideCount, currentIndex, setCurrentIndex } = context;
+
+  useTheme();
+
+  const dots = React.useMemo(
+    () =>
+      Array(slideCount)
+        .fill(null)
+        .map((_, index) => {
+          const firstDotIndex = Math.max(
+            0,
+            Math.min(
+              currentIndex - Math.ceil(maxCount / 2) + 1,
+              slideCount - maxCount,
+            ),
+          );
+          const lastDotIndex = Math.min(
+            slideCount - 1,
+            Math.max(currentIndex + Math.floor(maxCount / 2), maxCount - 1),
+          );
+
+          const isFirstSmallDot =
+            (index === firstDotIndex && index !== 0) ||
+            (index === lastDotIndex && index !== slideCount - 1);
+          const isSecondSmallDot =
+            (index === firstDotIndex + 1 && index !== 1) ||
+            (index === lastDotIndex - 1 && index !== slideCount - 2);
+
+          return (
+            <li key={index}>
+              <CarouselDot
+                aria-label={`Slide ${index}`}
+                isActive={index === currentIndex}
+                onClick={() => setCurrentIndex(index)}
+                isFirst={isFirstSmallDot}
+                isSecond={isSecondSmallDot}
+                isInvisible={index < firstDotIndex || index > lastDotIndex}
+              />
+            </li>
+          );
+        }),
+    [slideCount, maxCount, currentIndex, setCurrentIndex],
+  );
+
+  return (
+    <ol
+      className={cx('iui-carousel-navigation-dots', className)}
+      role='tablist'
+      {...rest}
+    >
+      {children ?? dots}
+    </ol>
+  );
+};
+
+const CarouselDot = (
+  props: {
+    isActive?: boolean;
+    isFirst?: boolean;
```

Comment:
`isFirst` and `isSecond` really looks strange. Maybe we should rename them to `small`, `medium` or `ellipsisStart`, `ellipsisEnd` or something better.

Reply:
I was just mirroring the names from CSS. I find your names to also be strange 😅 I already thought about `truncateStart`/`truncateEnd` and `truncateFirst`/`truncateSecond` but they are all strange